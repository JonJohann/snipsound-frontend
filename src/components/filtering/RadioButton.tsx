import React from 'react';
import { View, TextInput, TouchableOpacity, Text, StyleSheet } from 'react-native';
import { styles } from '../../styles/styles'

// Simple radiobutton
export default function RadioButton(props: any) {
    return (
        <TouchableOpacity
            onPress={() => props.callback(props.texts)}
            style={[(props.selected ? styles.active : styles.inactive), {marginHorizontal: 2}]}>
            <Text style={{fontFamily: 'Manrope_400Regular', fontSize: 20, color: (props.selected ? "black" : "grey")}}>
                {props.texts.charAt(0).toUpperCase() + props.texts.slice(1)}
                </Text>
        </TouchableOpacity>

    );
}