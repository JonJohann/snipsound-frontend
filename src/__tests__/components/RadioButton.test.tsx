import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import RadioButton from '../../components/filtering/RadioButton';
import Enzyme, { shallow }from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })

describe('<RadioButton />', () => {
    const callback = jest.fn();
    const selected = false;
    const texts = "Rap";

    const props = {
        callback,
        selected,
        texts
    }
    it('calls callback on click', () => {
        const _component = shallow(<RadioButton {...props}/>)
        _component.simulate('press')
        expect(callback.mock.calls.length).toEqual(1)
    })

    it('renders correctly', () => {
        const _component = renderer.create(<RadioButton {...props}/>).toJSON();
        expect(_component).toMatchSnapshot();
    });

});