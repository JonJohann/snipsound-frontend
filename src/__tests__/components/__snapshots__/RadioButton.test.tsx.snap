// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#e6e6e6",
      "borderColor": "#6d557000",
      "borderRadius": 10,
      "borderWidth": 1,
      "elevation": 2,
      "marginHorizontal": 2,
      "marginVertical": 5,
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "grey",
        "fontFamily": "Manrope_400Regular",
        "fontSize": 20,
      }
    }
  >
    Rap
  </Text>
</View>
`;
